src/zope/pagetemplate/tests/test_engine.py

```diff
@@ -86,6 +86,9 @@ def test_disallowed_builtin(self):
 class TestZopeContext(PlacelessSetup,
                       unittest.TestCase):
 
+    assertRaisesRegex = getattr(unittest.TestCase, 'assertRaisesRegex',
+                                getattr(unittest.TestCase, 'assertRaisesRegexp'))
```

Comment:
This is is always calling getattr(unittest.TestCase, 'assertRaisesRegexp').
Would not be better to:
```
try:
     assertRaisesRegex = unittest.TestCase.assertRaisesRegex
except Attribute Error:
     # Support X version of Y
     assertRaisesRegex = unittest.TestCase.assertRaisesRegexp
```

Not a blocker anyway.

Reply:
I write it that way basically to cheat: There's no implicit branch, all statements are always executed, and coverage is unaffected.

Reply:
I see your point :)

Reply:
Isn't this equivalent to 

```python
    assertRaisesRegex = getattr(unittest.TestCase, 'assertRaisesRegex',
                                unittest.TestCase.assertRaisesRegexp)
```

?

Why the second `getattr()`?

Reply:
Yes it is. I have no idea why I wrote it that way 😢  [In the past](https://github.com/zopefoundation/ZConfig/blob/master/ZConfig/tests/support.py#L63) I've written it directly.